src/Core/Context.php

```diff
@@ -0,0 +1,164 @@
+<?php
+/**
+ * Copyright 2017 OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace OpenCensus\Core;
+
+/**
+ * This class is an implementation of a generic context.
+ *
+ * Example:
+ * ```
+ * // Create and set a new context, which inherits values from the current
+ * // context and adds a new key/value pair of 'foo'/'bar'.
+ * $prev = Context::current()->withValue('foo', 'bar')->attach();
+ * try {
+ *   // Do something within the context
+ * } finally {
+ *   // This makes sure that $prev will be current after this execution
+ *   Context::current()->detach($prev);
+ * }
+ * // Here, we are 100% sure $prev is the current context.
+ * ```
+ */
+class Context
+{
+    /**
+     * @var Context
+     */
+    private static $current;
+
+    /**
+     * @var array
+     */
+    private $values;
+
+    /**
+     * Creates a new Context.
+     *
+     * @param array $initialValues
+     */
+    public function __construct($initialValues = [])
+    {
+        $this->values = $initialValues;
+    }
+
+    /**
+     * Creates a new context with the given key/value set.
+     *
+     * @param string $key
+     * @param mixed $value
+     * @return Context
+     */
+    public function withValue($key, $value)
```

Comment:
Do you need to call `withValue` multiple times if you want to add multiple key value pairs?

Reply:
Currently, yes. I can add a `withValues($keyValuesToMerge)`.

Reply:
Added.